### PR TITLE
test: add tests for gov proposal

### PR DESCRIPTION
**Pull request by hallazzang**
## Description

Add tests for `PublicPlanProposal`.

closes: #174

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Appropriate labels applied
- [x] Targeted PR against correct branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by dongsam** [comment]
It might be a little annoying, but it would be better if the test logics were added.

1. Verification that existing rewards are withdrawn without problems and no additional rewards are accumulated even after delete proposal passed or termination due to endtime modify proposal passed.
2. When the ratio and amount are changed in the existing staking object, whether the reward corresponding to the changed value is accumulated well.

**Comment by hallazzang** [comment]
Alright. I'll add more tests to cover that cases.

File: x/farming/keeper/proposal_handler_test.go [hunk 1]
```diff
@@ -12,6 +13,166 @@ import (
...
+	suite.Require().NoError(err)
+
+	plans = suite.keeper.GetPlans(suite.ctx)
+	suite.Require().Len(plans, 2)
+	plan = plans[1]
+	suite.Require().Equal(uint64(2), plan.GetId())
+	suite.Require().Equal("plan1", plan.GetName())
+}
+
+func testModifyPlaRequest(
```
Comment (kogisin) [inline x/farming/keeper/proposal_handler_test.go:73]:
```suggestion
func testModifyPlanRequest(
```

File: x/farming/keeper/keeper_test.go [hunk 2]
```diff
@@ -153,7 +157,7 @@ func (suite *KeeperTestSuite) AdvanceEpoch() {
 	suite.Require().NoError(err)
 }
 
-func (suite *KeeperTestSuite) SetFixedAmountPlan(id uint64, farmingPoolAcc sdk.AccAddress, stakingCoinWeightsMap map[string]string, epochAmountMap map[string]int64) {
+func (suite *KeeperTestSuite) CreateFixedAmountPlan(farmingPoolAcc sdk.AccAddress, stakingCoinWeightsMap map[string]string, epochAmountMap map[string]int64) {
```
Comment (kogisin) [inline x/farming/keeper/keeper_test.go:160]: Will we have any test cases that we need to pass the `PlanType` to create a fixed amount or a ratio plan? If we do, we might as well have the function to get the plan type as an argument.

**Review by kogisin — CHANGES_REQUESTED** [review]

File: x/farming/keeper/proposal_handler_test.go (hunk 1 above)
Comment (hallazzang) [reply x/farming/keeper/proposal_handler_test.go:73]: Thank you for finding it out!

File: x/farming/keeper/keeper_test.go (hunk 2 above)
Comment (hallazzang) [reply x/farming/keeper/keeper_test.go:160]: Currently, not. But maybe we should. Let me write tests as mentioned above by @dongsam and add `PlanType` parameter if needed.

Comment (hallazzang) [reply x/farming/keeper/keeper_test.go:169]: Those are going to be added in https://github.com/tendermint/farming/issues/219

Comment (kogisin) [reply x/farming/keeper/keeper_test.go:169]: Okay

**Review by dongsam — APPROVED** [review]
LGTM

**Comment by dongsam** [comment]
requested changes of @kogisin applied well, and he is on retreat, let me merge with skip approval

**Pull request merged**